cudaextender/samples/sample_cudaextender.cpp

```diff
@@ -0,0 +1,184 @@
+
+    // CSV SeedPairs file - Each row -> query_position_in_read_,
+    // target_position_in_read_
+    const std::string seed_pairs_file_path = std::string(CUDAEXTENDER_DATA_DIR) + "/sample_seed_pairs.csv";
+
+    //TODO - pinned seed_pairs
+    std::vector<SeedPair> h_seed_pairs;
+    // Following function loops through all seed_pairs in the sample_seed_pairs.csv and returns
+    // results in the passed vector
+    parse_seed_pairs(seed_pairs_file_path, h_seed_pairs);
+    std::cerr << "Number of Seed Pairs: " << h_seed_pairs.size() << std::endl;
+
+    // Define Scoring Matrix
+    const int32_t score_matrix[NUC2] = {91, -114, -31, -123, -1000, -1000, -100, -9100,
+                                        -114, 100, -125, -31, -1000, -1000, -100, -9100,
+                                        -31, -125, 100, -114, -1000, -1000, -100, -9100,
+                                        -123, -31, -114, 91, -1000, -1000, -100, -9100,
+                                        -1000, -1000, -1000, -1000, -1000, -1000, -1000, -9100,
+                                        -1000, -1000, -1000, -1000, -1000, -1000, -1000, -9100,
+                                        -100, -100, -100, -100, -1000, -1000, -100, -9100,
+                                        -9100, -9100, -9100, -9100, -9100, -9100, -9100, -9100};
+
+    // Allocate pinned memory for query and target strings
+    pinned_host_vector<int8_t> h_encoded_target(target_sequence.length());
+    pinned_host_vector<int8_t> h_encoded_query(target_sequence.length());
+
+    encode_sequence(h_encoded_target.data(), target_sequence.c_str(), target_sequence.length());
+    encode_sequence(h_encoded_query.data(), query_sequence.c_str(), query_sequence.length());
+    // Create a stream for async use
+    CudaStream stream0 = make_cuda_stream();
+    // Create an allocator for use with both APIs
+    const std::size_t max_gpu_memory = cudautils::find_largest_contiguous_device_memory_section();
+    DefaultDeviceAllocator allocator = create_default_device_allocator(max_gpu_memory);
+    // Reference for output
+    std::vector<ScoredSegmentPair> h_ssp;
+    // Create Ungapped Extender Object for both API modes
+    std::unique_ptr<Extender> ungapped_extender = create_extender(score_matrix,
+                                                                  NUC2,
+                                                                  xdrop_threshold,
+                                                                  input_no_entropy,
+                                                                  stream0.get(),
+                                                                  0,
+                                                                  allocator);
+    if (!device_ptr_api_mode)
+    {
+        ungapped_extender->extend_async(h_encoded_query.data(),
+                                        h_encoded_query.size(),
```

Comment:
Since the API accepts `int32_t`, please use `get_size<int32_t>(h_encoded_query)` (from `genomeworks/utils/signed_integer_utils.hpp`) instead of `.size()`.

Reply:
Done.

Reply:
Great.
Now, looking the code again, I guess it is a bit confusing to the reader when to use `get_size()`, when to use `.length()`/`.size()`. While the current usage is correct, I think it'd be better if we change all the `x.length()`/`x.size()` to `get_size(x)` (without the `<int32_t>`).

Reply:
Done.